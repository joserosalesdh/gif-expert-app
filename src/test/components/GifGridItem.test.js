import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifGridItem.js', () => {

    const title = 'Un titulo';
    const url = 'http://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    // beforeEach(() => {
    //     wrapper = shallow(<GifGridItem />);
    // })

    test('debe de mostrar el <GifGridItem /> correctamente ', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p').text().trim();

        expect(p).toBe(title);
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);

        expect(img.prop('alt')).toBe(title);

    })

    test('debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true)

    })




})



