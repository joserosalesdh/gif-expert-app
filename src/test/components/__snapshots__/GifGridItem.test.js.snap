// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js debe de mostrar el <GifGridItem /> correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
